src/plugins/data_explorer/public/components/sidebar/index.tsx

```diff
@@ -0,0 +1,120 @@
+import { setView } from '../../utils/state_management/metadata_slice';
+
+export const Sidebar: FC = ({ children }) => {
+  const { indexPattern: indexPatternId } = useTypedSelector((state) => state.metadata);
+  const dispatch = useTypedDispatch();
+  const [options, setOptions] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
+  const [selectedOption, setSelectedOption] = useState<EuiComboBoxOptionOption<string>>();
+  const { view, viewRegistry } = useView();
+  const views = viewRegistry.all();
+  const viewOptions = useMemo(
+    () =>
+      views.map(({ id, title }) => ({
+        value: id,
+        text: title,
+      })),
+    [views]
+  );
+
+  const {
+    services: {
+      data: { indexPatterns },
+      notifications: { toasts },
+    },
+  } = useOpenSearchDashboards<DataExplorerServices>();
+
+  useEffect(() => {
+    let isMounted = true;
+    const fetchIndexPatterns = async () => {
+      await indexPatterns.ensureDefaultIndexPattern();
+      const cache = await indexPatterns.getCache();
+      const currentOptions = (cache || []).map((indexPattern) => ({
+        label: indexPattern.attributes.title,
+        value: indexPattern.id,
+      }));
+      if (isMounted) {
+        setOptions(currentOptions);
+      }
+    };
+    fetchIndexPatterns();
+
+    return () => {
+      isMounted = false;
+    };
+  }, [indexPatterns]);
+
+  // Set option to the current index pattern
+  useEffect(() => {
+    if (indexPatternId) {
+      const option = options.find((o) => o.value === indexPatternId);
+      setSelectedOption(option);
+    }
+  }, [indexPatternId, options]);
+
+  return (
+    <EuiPageSideBar className="deSidebar" sticky>
+      <EuiSplitPanel.Outer className="eui-yScroll" hasBorder={true} borderRadius="none">
+        <EuiSplitPanel.Inner paddingSize="s" grow={false}>
+          <EuiComboBox
+            placeholder="Select a datasource"
+            singleSelection={{ asPlainText: true }}
+            options={options}
+            selectedOptions={selectedOption ? [selectedOption] : []}
+            fullWidth
+            onChange={(selected) => {
+              // TODO: There are many issues with this approach, but it's a start
+              // 1. Combo box can delete a selected index pattern. This should not be possible
+              // 2. Combo box is severely truncated. This should be fixed in the EUI component
+              // 3. The onchange can fire with a option that is not valid. discuss where to handle this.
+              // 4. value is optional. If the combobox needs to act as a slecet, this should be required.
+              const { value } = selected[0] || {};
+
+              if (!value) {
+                toasts.addWarning({
+                  id: 'index-pattern-not-found',
+                  title: i18n.translate('dataExplorer.indexPatternError', {
+                    defaultMessage: 'Index pattern not found',
```

Comment:
Wouldn't this be more user-friendly if it said "The index pattern `patten*` was not found" and maybe suggest that they recheck their spelling?

Reply:
The `if` condition here simply checks if there exists an index pattern so if this warning is shown, there exists no index pattern to show.